src/backend/src/services/slack.services.ts

```diff
@@ -0,0 +1,179 @@
+import UsersService from './users.services';
+import { getChannelName, getUserName } from '../integrations/slack';
+import AnnouncementService from './announcement.service';
+import { Announcement } from 'shared';
+import prisma from '../prisma/prisma';
+import { blockToMentionedUsers, blockToString } from '../utils/slack.utils';
+import { NotFoundException } from '../utils/errors.utils';
+
+/**
+ * Represents a slack event for a message in a channel.
+ */
+export interface SlackMessageEvent {
+  type: 'message';
+  subtype?: string;
+  channel: string;
+  event_ts: string;
+  channel_type: string;
+  [key: string]: any;
+}
+
+/**
+ * Represents a slack message event for a standard sent message.
+ */
+export interface SlackMessage extends SlackMessageEvent {
+  user: string;
+  client_msg_id: string;
+  text: string;
+  blocks: {
+    type: string;
+    block_id: string;
+    elements: any[];
+  }[];
+}
+
+/**
+ * Represents a slack message event for a deleted message.
+ */
+export interface SlackDeletedMessage extends SlackMessageEvent {
+  subtype: 'message_deleted';
+  previous_message: SlackMessage;
+}
+
+/**
+ * Represents a slack message event for an edited message.
+ */
+export interface SlackUpdatedMessage extends SlackMessageEvent {
+  subtype: 'message_changed';
+  message: SlackMessage;
+  previous_message: SlackMessage;
+}
+
+/**
+ * Represents a block of information within a message. These blocks with an array
+ * make up all the information needed to represent the content of a message.
+ */
+export interface SlackRichTextBlock {
+  type: 'broadcast' | 'color' | 'channel' | 'date' | 'emoji' | 'link' | 'text' | 'user' | 'usergroup';
+  range?: string;
+  value?: string;
+  channel_id?: string;
+  timestamp?: number;
+  name?: string;
+  unicode?: string;
+  url?: string;
+  text?: string;
+  user_id?: string;
+  usergroup_id?: string;
+}
+
+export default class slackServices {
+  /**
+   * Given a slack event representing a message in a channel,
+   * make the appropriate announcement change in prisma.
+   * @param event the slack event that will be processed
+   * @param organizationId the id of the organization represented by the slack api
+   * @returns an annoucement if an announcement was processed and created/modified/deleted
+   */
+  static async processMessageSent(event: SlackMessageEvent, organizationId: string): Promise<Announcement | undefined> {
+    //get the name of the channel from the slack api
+    const slackChannelName: string = (await getChannelName(event.channel)) ?? `Unknown_Channel:${event.channel}`;
+    const dateCreated = new Date(1000 * Number(event.event_ts));
```

Comment:
i thought we were letting the database just make this be the time that it gets created.